tests/test_expression.py

```diff
@@ -1276,3 +1286,16 @@ def test_expr_str() -> None:
                        3))"""
         ),
     )
+
+
+def test_expr_range_substituted() -> None:
+    expr = ValueRange(lower=First("Test"), upper=Sub(Last("Test"), Number(1)))
+
+    def func(expr: Expr) -> Expr:
+        if expr == First("Test"):
+            return Number(1)
+        if expr == Last("Test"):
+            return Number(11)
+        return expr
+
+    assert expr.substituted(func) == ValueRange(lower=Number(1), upper=Sub(Number(11), Number(1)))
```

Comment:
The class is called `ValueRange`, thus the name of the test should begin with `test_value_range_`. Please also move it to the correct location, there is already a test for `ValueRange`. It should be also tested, if `ValueRange` can be substituted as a whole expression.

Reply:
Fixed by https://github.com/Componolit/RecordFlux/pull/381/commits/0eb7b428aaeed454b876b131aa7108734b8bc643.